### DISCUSSION TITLE
How to save data between scenes?

### DISCUSSION THREAD

**Discussion by tfkfan** [excaliburjs/Excalibur#2256] [General]
I didn't found any correct(!) option to do that

**Comment by eonarheim** [comment]
Hi @tfkfan 

We don't have anything official at the moment.

What would you like to see in your game? It might help us design an API that'd serve your purpose.

In the past in our games we've used an external object that is known to all scenes (maybe even a singleton) to hold onto data between scenes.

@jedeen @kamranayub Any other suggestions?

**Reply by tfkfan** [reply]
Hi, so I've found a good solution, that's fine, thanks: 
```
  public onDeactivate(scene, nextScene) {
    if(nextScene instanceof GameScene)
      nextScene.data= this.data;
      }
  }
```

**Comment by KokoDoko** [comment]
You can use the `engine` reference to the main game to store your data :

```javascript
class Henk extends Actor {
    onInitialise(engine) {
       this.game = engine
   }
   updateScore(){
      this.game.score++
   }
}
```
Then in any other scene in any other actor you can get the score with
```javascript
class Wim extends Actor {
    onInitialise(engine) {
       this.game = engine
       console.log(this.game.score)
   }
}
```
